Tests/SonarScanner.Shim.Tests/PropertiesFileGeneratorTests.cs

```diff
@@ -113,6 +113,32 @@ public void FileGen_ValidFiles_SourceEncoding_Provided()
             Assert.IsTrue(settingsFileContent.Contains("sonar.sourceEncoding=test-encoding-here"), "Command line parameter 'sonar.sourceEncoding' is ignored.");
         }
 
+        [TestMethod]
+        public void FileGen_TFS_Coverage_Trx_Are_Written()
```

Comment:
Additional test cases: checking that user-supplied values take precedence?

Reply:
That's difficult to test (too many dependencies); I will refactor and add tests in a separate PR